cmake/PythonUtils.cmake

```diff
@@ -0,0 +1,67 @@
+# TODO: Eventually, we want to automatically turn on this auto-pip install functionality. That
+# should only be done when we KNOW that we are in an isolated configuration of Python.
+option(ENABLE_PIP_INSTALL "Allow automatic installation of Python dependency modules." OFF)
+
+function(python_test_modules)
+    cmake_parse_arguments(_ptm "DONT_INSTALL;REQUIRED" "REQUIREMENTS_FILE" "MODULES" ${ARGN})
+
+    if(NOT _ptm_MODULES)
+        message(FATAL_ERROR "python_test_modules() must be given a list of MODULES")
+    endif()
+
+    # Optional: attempt to install modules from pip before checking their existence.
+    if(ENABLE_PIP_INSTALL AND NOT _ptm_DONT_INSTALL)
+        if(NOT _PLASMA_ENSUREPIP_DONE)
+            message(STATUS "Bootstrapping pip...")
+            execute_process(
+                COMMAND "${Python3_EXECUTABLE}" -m ensurepip
+                RESULT_VARIABLE RETURNCODE
+                OUTPUT_QUIET ERROR_QUIET
+            )
+            if(RETURNCODE)
+                message(WARNING "Unable to bootstrap pip")
+            endif()
+            set(_PLASMA_ENSUREPIP_DONE ON CACHE INTERNAL "" FORCE)
```

Comment:
For clarity, I would probably use `YES` or `TRUE` rather than `ON`...  (CMake treats all of these, and others, as equivalent boolean constants -- see https://cmake.org/cmake/help/latest/command/if.html#condition-syntax)